File: 15-MQTT_Local_Server/pub.js
// MQTT publisher
var mqtt = require('mqtt')
var client = mqtt.connect('mqtt://localhost:1883')
var topic = 'initopic'
var message = '0'
var nambah = 0;

client.on('connect', ()=>{
    setInterval(()=>{
        nambah++;
        if (nambah >= 10) {
            nambah = 0
        }
        message = nambah.toString();
        client.publish(topic, message)
        console.log('Message sent!', message)
    }, 100)
})

File: 15-MQTT_Local_Server/pub.py
# publisher
import paho.mqtt.client as mqtt
from time import sleep

client = mqtt.Client()
client = connect('localhost', 1883)

x = 10

while True:
    x += 1
    if(x >= 20):
        x = 10
    print(x)
    sleep(0.01)
    client.publish('initopic_rpi', x)